src/mpid/ch4/src/ch4r_win.h

```diff
@@ -45,6 +45,108 @@ extern MPIR_T_pvar_timer_t PVAR_TIMER_rma_winlock_getlocallock ATTRIBUTE((unused
 extern MPIR_T_pvar_timer_t PVAR_TIMER_rma_wincreate_allgather ATTRIBUTE((unused));
 extern MPIR_T_pvar_timer_t PVAR_TIMER_rma_amhdr_set ATTRIBUTE((unused));
 
+MPL_STATIC_INLINE_PREFIX uint32_t MPIDI_CH4I_parse_info_accu_ops_str(const char *str)
+{
+    uint32_t ops = 0;
+    char *value, *token, *savePtr = NULL;
+
+    value = (char *) str;
+    token = (char *) strtok_r(value, ",", &savePtr);
+    while (token != NULL) {
+        MPIDI_CH4U_win_info_accu_op_shift_t op_shift;
+        /* if special value is set, skip others. */
+        if (!strncmp(token, "none", strlen("none"))) {
+            ops = 0;
+            break;
+        } else if (!strncmp(token, "any_op", strlen("any_op"))) {
+            /* add all ops */
+            for (op_shift = MPIDI_CH4I_ACCU_MAX_SHIFT;
```

Comment:
This sounds like a nitpicking, but I'm not too comfortable with assuming that `MAX_SHIFT` always comes first. Can't we just start from zero?

Reply:
Your comment makes sense. But it seems risky to me if we start from `MPIDI_CH4I_ACCU_OP_SHIFT_NONE`, e.g., if one changes the start value of the enum in future.

Reply:
resolved with another bit shift comment

Reply:
I think the probability of someone changing the order of enum values are higher than skipping the least significant bit (shift==0). That's why I'm advocating replacing `MPIDI_CH4I_ACCU_MAX_SHIFT` with `0` here. That way we're free to change the order of the values later (or insert another value in front of `MAX`).

Reply:
Both cases seem unlikely happen to me. We will never add new OPs unless MPI defines it. But I am OK to change it if you prefer.